### PR TITLE
Rework the use of lib: ["dom"]

### LINKED ISSUE #367: 2.0.3 version of ms-rest-js causes issues when we try to prepack sdks that take a dependency on that version
```bash
amarz-MBP:arm-kusto amarz$npm run prepack

> @azure/arm-kusto@2.0.1 prepack /Users/amarz/sdk/azure-sdk-for-js/sdk/kusto/arm-kusto
> npm install && npm run build

audited 72 packages in 0.576s
found 0 vulnerabilities


> @azure/arm-kusto@2.0.1 build /Users/amarz/sdk/azure-sdk-for-js/sdk/kusto/arm-kusto
> tsc -p tsconfig.json && rollup -c rollup.config.js && npm run minify

node_modules/@azure/ms-rest-js/es/lib/fetchHttpClient.d.ts:7:70 - error TS2304: Cannot find name 'RequestInit'.

7   abstract prepareRequest(httpRequest: WebResource): Promise<Partial<RequestInit>>;
                                                                       ~~~~~~~~~~~

node_modules/@azure/ms-rest-js/es/lib/fetchHttpClient.d.ts:9:25 - error TS2304: Cannot find name 'RequestInfo'.

9   abstract fetch(input: RequestInfo, init?: RequestInit): Promise<Response>;
                          ~~~~~~~~~~~

node_modules/@azure/ms-rest-js/es/lib/fetchHttpClient.d.ts:9:45 - error TS2304: Cannot find name 'RequestInit'.

9   abstract fetch(input: RequestInfo, init?: RequestInit): Promise<Response>;
                                              ~~~~~~~~~~~

node_modules/@azure/ms-rest-js/es/lib/fetchHttpClient.d.ts:9:67 - error TS2304: Cannot find name 'Response'.

9   abstract fetch(input: RequestInfo, init?: RequestInit): Promise<Response>;
                                                                    ~~~~~~~~

node_modules/@azure/ms-rest-js/es/lib/fetchHttpClient.d.ts:11:47 - error TS2304: Cannot find name 'Headers'.

11 export declare function parseHeaders(headers: Headers): HttpHeaders;
                                                 ~~~~~~~

node_modules/@azure/ms-rest-js/es/lib/nodeFetchHttpClient.d.ts:7:18 - error TS2304: Cannot find name 'RequestInfo'.

7     fetch(input: RequestInfo, init?: RequestInit): Promise<Response>;
                   ~~~~~~~~~~~

node_modules/@azure/ms-rest-js/es/lib/nodeFetchHttpClient.d.ts:7:38 - error TS2304: Cannot find name 'RequestInit'.

7     fetch(input: RequestInfo, init?: RequestInit): Promise<Response>;
                                       ~~~~~~~~~~~

node_modules/@azure/ms-rest-js/es/lib/nodeFetchHttpClient.d.ts:7:60 - error TS2304: Cannot find name 'Response'.

7     fetch(input: RequestInfo, init?: RequestInit): Promise<Response>;
                                                             ~~~~~~~~

node_modules/@azure/ms-rest-js/es/lib/nodeFetchHttpClient.d.ts:8:63 - error TS2304: Cannot find name 'RequestInit'.

8     prepareRequest(httpRequest: WebResource): Promise<Partial<RequestInit>>;
                                                                ~~~~~~~~~~~

node_modules/@azure/ms-rest-js/es/lib/webResource.d.ts:25:28 - error TS2304: Cannot find name 'Event'.

25     dispatchEvent: (event: Event) => boolean;
                              ~~~~~

node_modules/@azure/ms-rest-js/es/lib/webResource.d.ts:26:43 - error TS2304: Cannot find name 'Event'.

26     onabort: ((this: AbortSignalLike, ev: Event) => any) | null;
                                             ~~~~~

node_modules/@azure/ms-rest-js/es/lib/webResource.d.ts:27:77 - error TS2304: Cannot find name 'Event'.

27     addEventListener: (type: "abort", listener: (this: AbortSignalLike, ev: Event) => any, options?: any) => void;
                                                                               ~~~~~

node_modules/@azure/ms-rest-js/es/lib/webResource.d.ts:28:80 - error TS2304: Cannot find name 'Event'.

28     removeEventListener: (type: "abort", listener: (this: AbortSignalLike, ev: Event) => any, options?: any) => void;
                                                                                  ~~~~~


Found 13 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @azure/arm-kusto@2.0.1 build: `tsc -p tsconfig.json && rollup -c rollup.config.js && npm run minify`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the @azure/arm-kusto@2.0.1 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/amarz/.npm/_logs/2019-07-15T17_18_03_751Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @azure/arm-kusto@2.0.1 prepack: `npm install && npm run build`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the @azure/arm-kusto@2.0.1 prepack script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/amarz/.npm/_logs/2019-07-15T17_18_03_777Z-debug.log
amarz-MBP:arm-kusto amarz$ 
```

@kpajdzik - When I look at [`fetchHttpClient.ts`](https://github.com/Azure/ms-rest-js/blob/master/lib/fetchHttpClient.ts) there are no imports for `RequestInit`, `RequestInfo`, etc. because in `tsconfig.json` in `lib` we include `"dom"`. 
I am not sure why tsc complains about this when i am trying to build say `@azure/arm-kusto`.
- The quick fix bulb in vscode tries to import those types from `"node-fetch"`. Which is what we don't want since `fetchHttpClient` is the `abstract` base class that is common to both node and browser scenario.
- If I include `"dom"` in `"lib"` of `@azure/arm-kusto`'s tsconfig then the builld is fine. Don't quite like this solution. However, I am not sure of any other way to fix this.

/cc @bterlson, @daviwil - Any idea on how to fix this?

### PR DESCRIPTION
**Pull request by jeremymeng**
Porting changes from @azure/core-http PR
https://github.com/Azure/azure-sdk-for-js/pull/7500.

- Keep lib: ["dom"] for ./scripts/ as it depends on a dev common tool which
depends on v10 @azure/storage-blob which depends on an older version of
ms-rest-js. Attempting to remove it resulting in compile errors. This folder is
only used in dev/CI.  It's not part of the shipping library.

Resolves #367

### CODE REVIEW

**Comment by jeremymeng** [comment]
@xirzec @witemple-msft please take a look.

File: lib/dom.d.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/// <reference lib="dom" />
```
Comment (xirzec) [inline lib/dom.d.ts:4]: If we have dom-shim.d.ts, do we still need this lib reference directive? Or vice versa, given this do we need the shims? Feels like both is overkill.

Comment (jeremymeng) [reply lib/dom.d.ts:4]: It looks that we need both
- dom-shim.d.ts only include a couple of the types that exposed in our public api and we ship it in the npm package.
- internally we use more DOM types, e.g., in xhr http client. dom.d.ts is for internal usage.

I need to make sure that we don't pack this file in the released package though.

Comment (xirzec) [reply lib/dom.d.ts:4]: Gotcha! Thanks for explaining.

**Review by witemple-msft — APPROVED** [review]
LGTM. Is there any d.ts rollup in this repo? If so, we might need to disable that termporarily for the same reason that we had to disable it in core-http.

**Comment by jeremymeng** [comment]
> Is there any d.ts rollup in this repo?

No we don't use api-extractor in this repo.

File: lib/dom.d.ts (hunk 1 above)
Comment (jeremymeng) [reply lib/dom.d.ts:4]: @xirzec any other comments/questions, or this is good to merge?

Comment (xirzec) [reply lib/dom.d.ts:4]: I'm happy!

**Review by xirzec — APPROVED** [review]
LGTM :shipit:

**Pull request merged**